### DISCUSSION TITLE
.execute(raw query) transaction

### DISCUSSION THREAD

**Discussion by hubbusysyuhada** [mikro-orm/mikro-orm#2237] [General]
Hi, is there any way that I could do a transactions with raw query instead of entities using `.execute()` function mentioned in here (https://mikro-orm.io/docs/transactions/) ?

The case is, I want to create a "Membership" table, and if it's success, I want to insert row into "logs" table that contain some information. I tried to pass raw query with `BEGIN;` but it did nothing to my local pg.

the query is something like this

```
.execute(
`begin;
create new table "Membership";
insert into "logs" ("name") values ('create table "Membership"');
`)
```

if it succeed, i will run `.execute('commit;')` or `.execute('rollback;')` if error with try-catch statements

**Comment by B4nan** [comment]
In general you should not mix DDL and DML inside a single transaction. I believe it works in postgres, but e.g. mysql do not support this at all and will run things in separate transactions even if you do not state it in the queries. 

But technically what you can do is using `em.transactional()` and `em.execute()`:

```ts
await em.transactional(async () => {
  await em.execute('create new table "Membership"');
  await em.execute('insert into "logs" ("name") values (\'create table "Membership"\')');
});
```

note that to use the `em.execute()` method, the `em` variable here needs to be propertly typed as `SqlEntityManager` (which is always exported from driver packages aliased to just `EntityManager`)
